packages/markers/src/browser/problem/problem-decorator.ts

```diff
@@ -168,3 +171,131 @@ export namespace ProblemDecorator {
         (left.data.severity || Number.MAX_SAFE_INTEGER) - (right.data.severity || Number.MAX_SAFE_INTEGER);
 
 }
+
+@injectable()
+export class ProblemTabBarDecorator implements TabBarDecorator {
```

Comment:
Does it relate to navigator decorations? please put it in own file, see guidelines as well: https://github.com/theia-ide/theia/wiki/Coding-Guidelines#file-name

It should be easy to find a class with a file name, so putting many classes in one file is generally not welcomed. In this case these classes don't have anything in common, except `decorator` suffix in the name.

Reply:
Added https://github.com/theia-ide/theia/wiki/Coding-Guidelines#one-large-class-per-file

Reply:
I agree with @akosyakov, the class `ProblemTabBarDecorator` should be in it's own file.